Implement runbook demo api and endpoints

teferi opened the pull request.

maretskiy commented on ceagle/api_fake_data/fake_runbooks.py:
```diff
@@ -0,0 +1,93 @@
+def get_single_run():
+    return {
+        "_id": str(random.randint(1, 1000)),
+        "date": datetime.datetime.now(),
```
i guess that .isoformat() is missed

maretskiy replied:
Also, key name "date" looks inconsistent with the value, since value is date+time, however key "date" usually is used for YYY-dd-mm only

teferi replied:
true, also would replace them with started_at & finished_at

maretskiy commented on raml/abao_hooks.js:
```diff
@@ -85,3 +85,81 @@ hooks.before('GET /api/{version}/status/performance/{period} -> 200', function (
 });
+
+// runbooks
+var correct_runbook = {description: "test", name: "test", runbook: "test", type: "test"}
```
I would recommend to use same naming style within the file - mixedCase (like at L4-7):

    var bodyRunbook = { ... }  // correct runbook
    var bodyBadRunbook = { ... }

teferi replied:
done

maretskiy commented on raml/abao_hooks.js:
```diff
@@ -85,3 +85,81 @@ hooks.before('GET /api/{version}/status/performance/{period} -> 200', function (
+
+hooks.before('GET /api/{version}/region/{region}/runbooks/{runbook_id} -> 200', function (test, done) {
+  test.request.params = params;
+  test.request.params.runbook_id = "123"
```
Both run_id and runbook_id can be added once at L4

teferi replied:
done

maretskiy commented on tests/unit/test.py:
```diff
@@ -30,6 +30,19 @@ def setUp(self):
+
+    def delete(self, *args, **kwargs):
+        rv = self.app.delete(*args, **kwargs)
+        # there should be no content on delete method
```
This is strange, because fake_runbooks.py:74 returns returns "{}":

     flask.jsonify({}), 204

teferi replied:
looks like flask ignores any content for 204 responses. (NO_CONTENT =))

Cesarsar commented on raml/api.raml:
```diff
@@ -111,6 +111,100 @@ mediaType: application/json
               schema: !include schemas/404.json
               example: !include response_examples/404/region.json
 
+    /runbooks:
```
Could you please add response examples? At least for GET api calls with 200 response status.

teferi replied:
Done

teferi committed:
Add fake runbooks api and tests for it

teferi committed:
Add raml schema for runbook API

Includes schema, examples and abao hooks for testing

maretskiy commented on ceagle/api_fake_data/fake_runbooks.py:
```diff
@@ -0,0 +1,99 @@
+
+def get_single_run():
+    started_at = datetime.datetime.now().isoformat()
+    finished_at = (datetime.datetime.now() + datetime.timedelta(
```
Not critical for fake data, however this is strange to have finished_at in future.
It is more logical to have finished_at == ...now(), started_at = finished_at - timedelta(...)

teferi committed:
Use /runbook_runs instead of /runbooks/id/runs

This url is required to get a list of all runs and can be reused to
get list runs of a single runbook
Also add tags

Cesarsar commented on raml/response_examples/200/runbook_runs.json:
```diff
@@ -0,0 +1,84 @@
+			"finished_at": "2016-12-20T16:27:42.150227",
+			"output": "SGVsbG8gV29ybGQhCg==",
+			"return_code": 0,
+			"started_at": "2016-12-20T16:18:42.150142",
```
This time format differs of time that comes from /api/v1/status/availability/day, for example: "2016-12-22T00:00"
We need to use the same format.

teferi replied:
What you're saying is we don't need seconds and mseconds in the format? I don't think it's a good idea

teferi committed:
Update fake API responses in accordance with API changes

boris-42 commented:
@teferi please rebase this patch
